src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2588,7 +2588,7 @@ uint32_t RadioSpinel<InterfaceType, ProcessContextType>::Snprintf(char *aDest, u
     len = vsnprintf(aDest, static_cast<size_t>(aSize), aFormat, args);
     va_end(args);
 
-    return (len < 0) ? 0 : (static_cast<uint32_t>(len) > aSize - 1 ? aSize - 1 : static_cast<uint32_t>(len));
+    return (len < 0) ? 0 : Min(static_cast<uint32_t>(len), aSize - 1);
```

Comment:
May `Clamp` fit this case better?

Reply:
I agree `Clamp` fits this case. But I think the comparison of `len < 0` here is not for a numeric meaning. It's to indicate that some error happened. So I think we'd better keep it here. Thoughts?